monai/networks/utils.py

```diff
@@ -436,6 +448,51 @@ def copy_model_state(
     return dst_dict, updated_keys, unchanged_keys
 
 
+def save_state(
+    src: Union[torch.nn.Module, Dict],
+    path: PathLike,
+    create_dir: bool = True,
+    atomic: bool = True,
+    func: Optional[Callable] = None,
+    **kwargs,
+):
+    """
+    Save the state dict of input source data with PyTorch `save`.
+    It can save `nn.Module`, `state_dict`, a dictionary of `nn.Module` or `state_dict`.
+    And automatically convert the data parallel module to regular module.
+    For example::
+
+        save_state(net, path)
+        save_state(net.state_dict(), path)
+        save_state({"net": net, "opt": opt}, path)
+        net_dp = torch.nn.DataParallel(net)
+        save_state(net_dp, path)
+
+    Refer to: https://pytorch.org/ignite/v0.4.8/generated/ignite.handlers.DiskSaver.html.
+
+    Args:
+        src: input data to save, can be `nn.Module`, `state_dict`, a dictionary of `nn.Module` or `state_dict`.
+        path: target file path to save the state dict.
+        create_dir: whether to create dictionary of the path if not existng, default to `True`.
+        atomic: if `True`, state is serialized to a temporary file first, then move to final destination.
+            so that files are guaranteed to not be damaged if exception occurs.
+        func: the function to save file, if None, default to `torch.save`.
+        kwargs: other args for the save `func` except for the checkpoint and filename.
+            default `func` is `torch.save()`, details of other args:
+            https://pytorch.org/docs/stable/generated/torch.save.html.
+
+    """
+
+    ckpt: Dict = {}
+    if isinstance(src, dict):
+        for k, v in src.items():
+            ckpt[k] = get_state_dict(v)
+    else:
+        ckpt = get_state_dict(src)
+
+    save_obj(obj=ckpt, path=path, create_dir=create_dir, atomic=atomic, func=func, **kwargs)
```

Comment:
looks like all the keywords arguments except for `ckpt` could be merged into `kwargs`

Reply:
updated in the latest commit.
I kept `obj` and `path` as these 2 args don't have default values, users must provide them.

Thanks.